md-setup/src/main/java/com/wipro/ats/bdre/md/setup/SetupDB.java

```diff
@@ -112,17 +114,17 @@ public static String replaceVars(String line) throws Exception {
                 m.appendReplacement(sb, map.get(key).toString());
             }
             else{
-                throw new Exception("There is no variable defined for ${"+key+"} for line: "+line);
+                throw new MetadataException("There is no variable defined for ${"+key+"} for line: "+line);
             }
         }
         m.appendTail(sb);
 
         return sb.toString();
     }
-    private String[] getColumns(String line) throws Exception {
+    private String[] getColumns(String line) throws MetadataException {
         if (line.trim().isEmpty() || line.trim().startsWith("--") || line.trim().startsWith("#")) {
             LOGGER.info("Ignoring comment:" + line);
-            return null;
+            return new String[]{};
         }
         //replace the variables
         line=replaceVars(line);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "line". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1226)